client/client.go

```diff
@@ -1,3 +1,65 @@
+/*
+Package client implements everything required for interacting with a notary repository.
+
+Usage
+
+Use this package by creating a new repository object and calling methods on it.
+
+	package main
+
+	import (
+		"net/http"
+		"os"
+		"path/filepath"
+
+		notary "github.com/docker/notary/client"
+		"github.com/docker/notary/cryptoservice"
+		"github.com/docker/notary/storage"
+		"github.com/docker/notary/trustmanager"
+		"github.com/docker/notary/trustpinning"
+	)
+
+	func main() {
+		rootDir := ".trust"
+		if err := os.MkdirAll(trustDir, 0700); err != nil {
+			panic(err)
+		}
+
+		metadata, err := storage.NewSimpleFileStore(filepath.Join(rootDir, "metadata"), "json")
+		if err != nil {
+			panic(err)
+		}
+
+		remote, err := storage.NewHTTPStore(server.URL, "metadata", "json", "key", http.DefaultTransport)
+		if err != nil {
+			panic(err)
+		}
+
+		cryptoService := cryptoservice.NewCryptoService(trustmanager.NewKeyMemoryStore(demoPassRetriever))
+
+		repo, err := notary.NewNotaryRepository(
```

Comment:
If you prefer, this can also be changed to:
```
notary.NewFileCachedNotaryRepository(
  rootDir,
  "docker.io/library/alpine",
  "https://notary.docker.io",
  roundTripper, // note this is different from remote
  demoPassRetrieverType, // note this needs to be wrapped
  trustpinning.TrustPinConfig{},
)
```

and then there's no need for the `metadata`, changelist, cryptoservice, or other `filepath.Join`.  That said, it will require some changes for the roundtripper and passretriever

Reply:
This can't use the (vanilla) default http transport for the round tripper?

What does it mean that the pass retriever needs to be wrapped?

Reply:
> This can't use the (vanilla) default http transport for the round tripper?

It likely won't because of Hub's token auth, but it's worth a try to confirm?  @tiborvass and I rigged up a challenge-response round tripper to get tags listing, as Hub will require an anonymous token for public repros.

> What does it mean that the pass retriever needs to be wrapped?

You'll need to pass a passretriever type, but actually it seems you already have satisfied the interface :+1:

Reply:
Ok, the example is updated and added a new function to generate the right round tripper for this to work correctly.

Reply:
```
go run notary.go
2.7	9f08005dff552038f0ad2f46b8e65ff3d25641747d3912e3ea8da6785046561a
3.3	8fd4b76819e1e5baac82bd0a3d03abfe3906e034cc5ee32100d12aaaf3956dc7
2.6	9ace551613070689a12857d62c30ef0daa9a376107ec0fff0e34786cedb3399b
latest	1072e499f3f655a032e88542330cf75b02e7bdf673278f701d7ba61629ee3ebe
3.1	d9477888b78e8c6392e0be8b2e73f8c67e2894ff9d4b8e467d1488fcceec21c8
3.2	19826d59171c2eb7e90ce52bfd822993bef6a6fe3ae6bb4a49f8c1d0a01e99c7
3.5	af2a5bd2f8de8fc1ecabf1c76611cdc6a5f1ada1a2bdd7d3816e121b70300308
edge	79d50d15bd7ea48ea00cf3dd343b0e740c1afaa8e899bee475236ef338e1b53b
3.4	833ad81ace8277324f3ca8c91c02bdcf1d13988d8ecf8a3f97ecdd69d0390ce9
3.6	1072e499f3f655a032e88542330cf75b02e7bdf673278f701d7ba61629ee3ebe
integ-test-base	3952dc48dcc4136ccdde37fbef7e250346538a55a0366e3fccc683336377e372
```